bindings/pyroot_experimental/pythonizations/python/ROOT/_numbadeclare.py

```diff
@@ -0,0 +1,278 @@
+            raise Exception('Failed to jit Python wrapper with numba.cfunc')
+        func.__py_wrapper__ = pywrappercode
+        func.__numba_cfunc__ = nbcfunc
+
+        # Get address of jitted wrapper function
+        address = nbcfunc.address
+
+        # Infer name of the C++ wrapper function
+        if not name:
+            name = func.__name__
+
+        # Build C++ wrapper for jitting with cling
+
+        # Define input signature
+        input_types_ref = ['ROOT::{}&'.format(t) if 'RVec' in t else t for t in input_types]
+        input_signature = ', '.join('{} x_{}'.format(t, i) for i, t in enumerate(input_types_ref))
+
+        # Define function pointer types
+        func_ptr_input_types = []
+        for t in input_types:
+            if 'RVec' in t:
+                innert = get_inner_type(t)
+                if innert == 'bool':
+                    # Special treatment for bool: In numpy, bools have 1 byte
+                    func_ptr_input_types += ['char*, int']
+                else:
+                    func_ptr_input_types += ['{}*, int'.format(innert)]
+            else:
+                func_ptr_input_types += [t]
+        if 'RVec' in return_type:
+            # See C++ wrapper code for the reason using these types
+            innert = get_inner_type(return_type)
+            func_ptr_input_types += ['{}**, long*, int'.format('char' if innert == 'bool' else innert)]
+        func_ptr_type = '{RETURN_TYPE}(*)({INPUT_TYPES})'.format(
+                RETURN_TYPE='void*' if 'RVec' in return_type else return_type,
+                INPUT_TYPES=', '.join(func_ptr_input_types)
+                )
+
+        # Define function call
+        vecbool_conversion = []
+        func_args = []
+        for i, t in enumerate(input_types):
+            if 'RVec' in t:
+                func_args += ['x_{0}.data(), x_{0}.size()'.format(i)]
+                if get_inner_type(t) == 'bool':
+                    # Copy the RVec<bool> to a RVec<char> to match the numpy memory layout
+                    func_args[-1] = func_args[-1].replace('x_', 'xb_')
+                    vecbool_conversion += ['ROOT::RVec<char> xb_{0} = x_{0};'.format(i)]
+            else:
+                func_args += ['x_{}'.format(i)]
+        if 'RVec' in return_type:
+            # See C++ wrapper code for the reason using these arguments
+            func_args += ['&ptr, &size, dtype_size']
+
+        # Define return operation
+        if 'RVec' in return_type:
+            innert = get_inner_type(return_type)
+            if innert == 'bool': innert = 'char'
+            return_op = '\n    '.join([
+                '// Because an RVec cannot take the ownership of external data, we have to copy the returned array',
+                'long size; // Size of the returned array',
+                '{}* ptr; // Pointer to the data of the returned array'.format(innert),
+                'int dtype_size = sizeof({}); // Size of array datatype in bytes'.format(innert),
```

Comment:
nitpicking, should be `std::size_t` instead of `int`

Reply:
Correct :) Other questions: Is `long` enough for the size or should we go do `unsigned long`? Or even `unsigned long long`? Or just a `size_t`? :D

Reply:
we should go with the type of `sizeof`, `std:size_t`

Reply:
we cast it anyway to an `int` when we call the function. and which type should be larger than an `int`?

Reply:
alright!